src/coreComponents/common/TypeDispatch.hpp

```diff
@@ -277,7 +342,38 @@ bool dispatch( TypeList< Ts... > const types,
   {
     GEOS_ERROR( "Type " << LvArray::system::demangle( type.name() ) << " was not dispatched.\n" <<
                 "Check the stack trace below and revise the type list passed to dispatch().\n" <<
-                "If you are unsure about this error, please report it to GEOSX issue tracker." );
+                "If you are unsure about this error, please report it to GEOS issue tracker." );
+  }
+  return success;
+}
+
+
+/**
+ * @brief
+ *
+ * @tparam LIST
+ * @tparam LAMBDA
+ * @tparam Ts
+ * @param combinations
+ * @param lambda
+ * @param objects
+ * @return true
+ * @return false
+ */
+template< typename LIST, typename LAMBDA, typename ... Ts >
+bool dispatchCombinations( LIST const combinations,
+                           LAMBDA && lambda,
+                           Ts && ... objects )
+{
+  bool const success = internal::dispatchViaTable( combinations,
+                                                   std::make_tuple( std::type_index( typeid(objects))... ),
```

Comment:
`createTypeIndexTuple( combinations )`?

Reply:
Since we're using a variadic parameter here necessitating it is the last parameter, can we restructure the downstream calls to order the arguments similarly? (i.e. lambda 2nd instead of last, type tuple list last in place of `objects` since we process `objects` to get that argument)?